keps/sig-scheduling/5027-dra-admin-controlled-device-attributes/README.md

```diff
@@ -0,0 +1,848 @@
+
+## Motivation
+
+### Goals
+
+- Enable [admin-controlled](../5055-dra-device-taints-and-tolerations) device taints.
+
+- Enable updating how devices are seen in the cluster without having to use
+  driver-specific APIs which influence what a driver puts into ResourceSlices.
+
+### Non-Goals
+
+- At least for alpha: extend `kubectl` to provide a unified view of devices
+  together with all overrides that apply to them.
+
+## Proposal
+
+The intent to override device attributes must be recorded persistently so that
+it is preserved even when a ResourceSlice gets removed or updated. To achieve
+this, a new cluster-scoped ResourceSliceOverride type gets added. A single
+ResourceSliceOverride object specifies device attributes that apply to all
+devices matching a CEL expression, i.e. the same way as users select devices in
+a ResourceClaim.
+
+The scheduler must merge these additional attributes with the ones provided by
+the DRA drivers on-the-fly while it gathers information about available
+devices.
+
+### Notes/Constraints/Caveats
+
+Users who look at ResourceSlices to figure out which devices are available also
+need to consider ResourceSliceOverrides to get the full picture. Copying from
+the ResourceSliceOverride spec into the ResourceSlice status could help here,
+but would not be instantaneous and potentially cause write amplification (one
+ResourceSliceOverride affecting many different devices) and therefore is not
+part of this proposal.
+
+Perhaps `kubectl describe resourceslices` can be extended to include the
+additional information. For now this is out of scope.
+
+Creating a ResourceSliceOverride is racing with on-going scheduling attempts,
+which is unavoidable.
+
+### Risks and Mitigations
+
+From a security perspective, permission to override device attributes is
+expected to be limited to privileged users who already have the ability to add
+or remove DRA drivers, so there won't be a substantial difference. Normal users
+can only decide for themselves whether they ignore taints and do so at their
+own risk. They cannot break other users' workloads.
+
+Performance in the scheduler could be an issue. This will be mitigated by
+(re-)applying overrides only when they or the device definitions change, which
+should be rare.
+
+## Design Details
+
+### API
+
+The ResourceSliceOverride is a cluster-scoped type in the `resource.k8s.io` API
+group, initially in `v1alpha3` (the alpha version in Kubernetes 1.32). Because
+it may be useful to clean up after disabling the feature and because the
+device taint feature also uses this type, it gets served unconditionally as long as
+the `v1alpha3` version is enabled. Fields related specifically to this KEP
+are feature-gated.
+
+```Go
+type ResourceSliceOverride struct {
+metav1.TypeMeta
+    // Standard object metadata
+    // +optional
+    metav1.ObjectMeta
+
+    // Changing the spec automatically increments the metadata.generation number.
+    Spec ResourceSliceSpec
```

Comment:
nit: this should be `ResourceSliceOverrideSpec`, no?

Reply:
Yes, fixed.